halo2-base/src/gates/flex_gate.rs

```diff
@@ -885,7 +1096,11 @@ impl<F: ScalarField> GateInstructions<F> for GateChip<F> {
         ctx.get(-5)
     }
 
-    // returns little-endian bit vectors
+    /// Constrains and returns little-endian bit vector representation of `a`.
+    /// 
+    /// Assumes `range_bits <= number of bits in a`.
```

Comment:
the constraints will fail if `range_bits < number of bits in a`

it actually assumes `range_bits >= number of bits in a`, but not too much greater since it depends on `.to_repr()`